### PR TITLE
Recommend use of repeatable payloads

### PR DESCRIPTION
**Pull request by hisnamewasjaan**
## Uploading
Added extra info from @nacx recommending the use of repeatable payloads.
## Downloading
Removed some deprecated and non-existing api from description,

### CODE REVIEW

File: start/blobstore.md [hunk 1]
```diff
@@ -392,9 +392,10 @@ an object from a container that does not exist is a state problem, and should th
 <!--update to use ByteSource -->
 
 ### Uploading 
-As long as you use either `InputStream` or `File` as the payload for your blob, you should
+As long as you use either `ByteSource` or `File` as the payload for your blob, you should
 be fine. Note that in S3, you must calculate the length ahead of time, since it doesn't support
 chunked encoding.
+It is usually better to use a repeatable Payload like ByteSource instead of InputStream, since this allows parallel uploads and retrying on errors.
```
Comment (gaul) [inline start/blobstore.md]: Could you annotate as `ByteSource` and `InputStream`?

**Comment by gaul** [comment]
It would also be great to add a similar blurb in `BlobBuilder.java`.

File: start/blobstore.md (hunk 1 above)
Comment (hisnamewasjaan) [reply start/blobstore.md]: Sorry, I dont follow? Annotate what?

Comment (hisnamewasjaan) [reply start/blobstore.md]: Got it, thx :) 
Commit amended.

**Comment by gaul** [comment]
Thank you for your contribution @hisnamewasjaan!

**Pull request merged**